Neuro/NeuroPlugin.cs

```diff
@@ -15,194 +21,24 @@ namespace Neuro;
-                pathIndex++;
-                if (pathIndex > currentPath.Length - 1)
-                {
-                    pathIndex = currentPath.Length - 1;
-                    nextWaypoint = currentPath[pathIndex];
-                    break;
-                }
-
-                nextWaypoint = currentPath[pathIndex];
-            }
-
-            directionToNearestTask = (nextWaypoint - (Vector2)PlayerControl.LocalPlayer.GetTruePosition()).normalized;
-
-
-            LineRenderer renderer = arrow;
-            renderer.SetPosition(0, PlayerControl.LocalPlayer.GetTruePosition());
-            renderer.SetPosition(1, PlayerControl.LocalPlayer.GetTruePosition() + directionToNearestTask);
-            renderer.widthMultiplier = 0.1f;
-            renderer.positionCount = 2;
-            renderer.startColor = Color.red;
-        }
-        else
-        {
-            directionToNearestTask = Vector2.zero;
-        }
-
-        return true;
-    }
-
-    public void MeetingBegin()
-    {
-        Debug.Log("NEURO: MEETING CALLED");
-        vision.ReportFindings();
-    }
-
-    public void MeetingEnd()
-    {
-        Debug.Log("NEURO: MEETING IS FINISHED");
-        vision.MeetingEnd();
-    }
-    public IEnumerator EvaluatePath(NormalPlayerTask initial)
-    {
-        currentPath = pathfinding.FindPath(PlayerControl.LocalPlayer.transform.position, initial.Locations[0]);
-        pathIndex = 0;
-
-        while (true)
-        {
-            yield return new WaitForSeconds(1);
-
-            PlayerTask task = PlayerControl.LocalPlayer.myTasks[0];
-
-            PlayerTask nextTask = null;
-            if (task.IsComplete)
-            {
-                Debug.Log("Task is complete");
-                foreach (PlayerTask t in PlayerControl.LocalPlayer.myTasks)
-                {
-                    if (!t.IsComplete && t.HasLocation)
-                    {
-                        nextTask = t;
-                        Debug.Log(nextTask.name);
-                        break;
-                    }
-                }
-            }
-            else
-            {
-                nextTask = task;
-            }
-            if (nextTask != null)
-            {
-                Debug.Log("Next task isn't null");
-                currentPath = pathfinding.FindPath(PlayerControl.LocalPlayer.transform.position, nextTask.Locations[0]);
-                pathIndex = 0;
-
-                //pathfinding.DrawPath(currentPath);
-            }
-        }
+        Harmony.CreateAndPatchAll(Assembly.GetExecutingAssembly(), Id);
+
+        MinigamesHandler = new MinigamesHandler();
+        MovementHandler = new MovementHandler();
+        PathfindingHandler = new PathfindingHandler();
+        RecordingHandler = GameObjectUtilities.CreatePermanentSingleton<RecordingHandler>();
+        TasksHandler = GameObjectUtilities.CreatePermanentSingleton<TasksHandler>();
+        VisionHandler = GameObjectUtilities.CreatePermanentSingleton<VisionHandler>();
```

Comment:
GameObjects created here get deleted despite DontDestroyOnLoad, and their update functions aren't called
There's AddComponent method inside BasePlugin that can you can use instead:
```
RecordingHandler = AddComponent<RecordingHandler>();
TasksHandler = AddComponent<TasksHandler>();
VisionHandler = AddComponent<VisionHandler>();
```